jjb/dynamic/scale-ci-ms-rosa.yml

```diff
@@ -36,14 +36,9 @@
             echo "INFO: Installing rpm requirements..."
             sudo /usr/bin/yum -y install python3 python3-pip golang-bin make
             echo "INFO: Installing python requirements..."
-            mkdir -p ${REMOTE_WORKSPACE}
-            if [[ ! -d "${REMOTE_WORKSPACE}/perfscale-managed-services" ]] ; then
-              git clone https://github.com/cloud-bulldozer/perfscale-managed-services.git ${REMOTE_WORKSPACE}/perfscale-managed-services
-            else
-              cd ${REMOTE_WORKSPACE}/perfscale-managed-services
-              remote=\$(git remote -v | grep "cloud-bulldozer/perfscale-managed-services.git (fetch)" | cut -f 1)
-              git pull \${remote} main
-            fi
+            mkdir -p ${REMOTE_WORKSPACE}/run-${RUN_FOLDER}
+            echo "INFO: Using ${REMOTE_WORKSPACE}/run-${RUN_FOLDER} as working directory on Jump host ${JUMP_HOST}"
+            git clone https://github.com/cloud-bulldozer/perfscale-managed-services.git ${REMOTE_WORKSPACE}/run-${RUN_FOLDER}/perfscale-managed-services
```

Comment:
Maybe deleting the repo in the dir if exists before cloning? This way we can avoid having multiple directories in the folder since it's using the timestamp for the repo to be unique (   export RUN_FOLDER=$(date +%Y%m%d-%H%M%S) ). Thoughts?

Reply:
I prefer not to add rm commands on scripts, when they are managed by vars, to avoid massive deletions if anything fails on create the var content.

I think it is better to always clone, keep all the required code inside the same dir and do housekeeping periodically

Reply:
+1 for adding it into the new dir. especially if we run multiple tests from the same JH we dont want to have any collisions.

Reply:
I think it's just a matter of enabling "Delete workspace before build starts." in case of using jenkins.

Reply:
workspace is on jenkins host, no on the jump host

Reply:
jenkins--

Reply:
Maybe add a check and delete just the repo that gets cloned if it exists since we want to have latest copy for each run? Storing the previous copies might be redundant IMHO. Thoughts?

Reply:
Working folders are always unique, as they includes date and time to seconds, so no verification is required.

Any option including deleting repo will cause concurrent executions non stable, so even we can consume a bit of disk space, I still prefer to checkout each time by now. In the future, if we could store the scripts as packages on github or containers or something, we will reconsider this stuff, but at this moment, the only solution I can confirm the code I'm using, is to download each time I needed